include/libdnf/base/goal_elements.hpp

```diff
@@ -95,6 +96,21 @@ enum class GoalAction {
 enum class GoalSetting { AUTO, SET_TRUE, SET_FALSE };
 enum class GoalUsedSetting { UNUSED, USED_TRUE, USED_FALSE };
 
+struct GroupJobSettings {
+    bool ignore_case{false};
+    bool with_group_id{false};
+    bool with_group_name{false};
+
+    bool with_optional{false};
```

Comment:
Groups uses I thing four types of packages - `mandatory`, `default`, `optional`, `conditional`. In DNF on API it was possible to handle each of those types separatelly or all together therefore I am not sure whether handling only optional is sufficient. I know that many applications reduce the behavior only to two groups - default dnf set plus optional.

Reply:
Right. I'll change the `bool with_optional` to `libdnf::comps::PackageType package_types` defaulting to `MANDATORY | DEFAULT | CONDITIONAL`